crates/rrg/src/action/dump_process_memory.rs

```diff
@@ -207,12 +194,37 @@ mod linux {
         }
     }
 
+    /// Allows to read the contents of a running process' memory.
+    pub struct ReadableProcessMemory {
+        mem_file: File,
+    }
+
     impl ReadableProcessMemory {
         /// Opens the contents of process `pid`'s memory for reading.
         pub fn open(pid: u32) -> std::io::Result<Self> {
             let mem_file = File::open(format!("/proc/{pid}/mem"))?;
             Ok(Self::from_file(mem_file))
         }
+
+        pub fn from_file(mem_file: File) -> Self {
+            Self { mem_file }
+        }
+    }
+
+    impl MemoryReader for ReadableProcessMemory {
+        /// Reads a slice `\[offset..(offset+length)\]` of the opened process' memory
+        /// and returns it as a [`Vec<u8>`]. `offset` is considered as an absolute offset
+        /// in the process' address space. If the slice falls outside the memory's address space,
+        /// the returned buffer will be truncated.
```

Comment:
No need to copy doc comment if this is an implementation of a trait method (unless it adds some implementation-specific information).